setup.py

```diff
@@ -300,7 +300,8 @@ def check_linker_need_libatomic():
 asm_files = []
 
 asm_key = ''
-if BUILD_WITH_BORING_SSL_ASM and not BUILD_WITH_SYSTEM_OPENSSL:
+if (BUILD_WITH_BORING_SSL_ASM.upper() not in ['FALSE', '0', ''] and
```

Comment:
Fair enough, but as pointed out in https://github.com/grpc/grpc/issues/24498#issuecomment-778123526 there are plenty of other options that can be set for setup.py  and by changing the behavior for BUILD_WITH_BORING_SSL_ASM only, we're basically introducing an inconsistency (and having some setup.py with one behavior and some with a different behavior is something we really don't want).

My suggestion is to introduce a private function e.g. `_get_build_option_bool_value(env_name))` that has the `upper() not in ['FALSE', '0', '']` logic in it and then use it for all the boolean build options in setup.py - there are quite a few.